### ISSUE TITLE
Unexpected LTE datetime range rounding with `now/d-0d` and other `-0` queries

### ISSUE THREAD

**Issue by Dzuelu** [elastic/elasticsearch#77896]
**Elasticsearch version** (`bin/elasticsearch --version`): Elasticsearch 7.1

**Plugins installed**: None

**JVM version** (`java -version`): Default for AWS::Elasticsearch::Domain

**OS version** (`uname -a` if on a Unix-like system): Default for AWS::Elasticsearch::Domain

**Description of the problem including expected versus actual behavior**: Subtracting 0d from now/d should not change the date. We use this for `0-30` and `30-60` date queries and is only an issue when using `-0d`. 

**Steps to reproduce**: 

 1. Create an index with a date field datatype
 2. ```
    GET /offering-index-123/_search
    {
      "explain": true, 
      "query": {
        "bool": {
          "must": [
            {
              "range": {
                "purchaseDate": {
                  "lte": "now/d-0d",
                  "gt": "now/d-30d"
                }
              }
            }
          ]
        }
      }
    }
    ```
 3. Grab the last range value and convert to iso timestamp to see clearly
    ```
    {
      "value" : 1.0,
      "description" : "purchaseDate:[... TO 1631836799998]",
      "details" : [ ]
    }
    ```

I did this for many values today to get these values on 2021-09-16:

"lte": "now/d-0y"
`2022`-09-16T23:59:59.998Z

"lte": "now/d-0M"
2021-`10`-16T23:59:59.998Z

"lte": "now/d-0d"
2021-09-`17`T23:59:59.998Z

"lte": "now-0d/d"
2021-09-17T23:59:59.999Z

"lte": "now/d-0h"
2021-09-17T`00`:59:59.998Z

"lte": "now/d-0m"
2021-09-17T00:`00`:59.998Z

"lte": "now/d-0s"
2021-09-17T00:00:`00`.998Z

"lte": "now/d" // The value we actually want for -0d
2021-09-16T23:59:59.999Z

"lte": "now/d+0d" // -0d and +0d make the same wrong value
2021-09-17T23:59:59.998Z

**Comment by ywelsch** [comment]
Have you tried reproducing this in any recent versions of Elasticsearch?

In particular, https://github.com/elastic/elasticsearch/pull/43303 looks like a fix for this.

**Comment by Dzuelu** [comment]
If that was included in v7.2.1, looks like were one minor version behind the fix. Thanks, I'll close this then!
```
"version" : {
    "number" : "7.1.1",
    "build_flavor" : "oss",
    "build_type" : "tar",
    "build_hash" : "0788933",
    "build_date" : "2020-07-30T22:09:03.152410Z",
    "build_snapshot" : false,
    "lucene_version" : "8.0.0",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  }
```
